tunnel_routes.go

```diff
@@ -0,0 +1,137 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// TunnelRoute is the full record for a route.
+type TunnelRoute struct {
+	Network    string    `json:"network"`
+	TunnelId   string    `json:"tunnel_id"`
+	TunnelName string    `json:"tunnel_name"`
+	Comment    string    `json:"comment"`
+	CreatedAt  time.Time `json:"created_at"`
+	DeletedAt  time.Time `json:"deleted_at"`
+}
+
+// tunnelRouteListResponse is the API response for listing tunnel routes.
+type tunnelRouteListResponse struct {
+	Response
+	Result []TunnelRoute `json:"result"`
+}
+
+type tunnelRouteResponse struct {
+	Response
+	Result TunnelRoute `json:"result"`
+}
+
+func extractTunnelRouteResponse(responseBody []byte, err error) (TunnelRoute, error) {
+	if err != nil {
+		return TunnelRoute{}, err
+	}
+
+	var routeResponse tunnelRouteResponse
+	err = json.Unmarshal(responseBody, &routeResponse)
+	if err != nil {
+		return TunnelRoute{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return routeResponse.Result, nil
+}
+
+// TunnelRoutes lists all defined routes for tunnels in the account.
+//
+// See: https://api.cloudflare.com/#tunnel-route-list-tunnel-routes
+func (api *API) TunnelRoutes(ctx context.Context) ([]TunnelRoute, error) {
```

Comment:
this method should take into consideration the [`PaginationOptions`](https://github.com/cloudflare/cloudflare-go/blob/365afd256caf70673bc294886b1eeb08cddae7ca/cloudflare.go#L458-L461) for when the number of routes is over the default pager.

```suggestion
func (api *API) TunnelRoutes(ctx context.Context, pageOpts PaginationOptions) ([]TunnelRoute, ResultInfo, error) {
```

you'll also need to factor these in accordingly in the method below.